client/src/main/java/org/eclipse/hono/client/CommandResponse.java

```diff
@@ -88,7 +88,8 @@ public static CommandResponse from(
             return null;
         } else {
             try {
-                final boolean addDeviceIdToReply = "1".equals(requestId.substring(0, 1));
+                final String replyToOptionsBitFlag = requestId.substring(0, 1);
+                final boolean addDeviceIdToReply = Command.isReplyToContainedDeviceIdOptionSet(replyToOptionsBitFlag);
```

Comment:
we now have a mutual dependency between `Command` and `CommandResponse` ...

Reply:
The `Command` class depends on `CommandResponse` only because of the reference in the `getDeviceFacingReplyToId` method javadoc (pointing to `CommandResponse#getReplyToId`).
I put that reference in there some time ago mainly to connect the two places that dealt with this bit flag logic.
Now that this logic is handled by common methods, the reference isn't needed so much anymore, I can remove it.